python/whylogs/api/logger/experimental/logger/actor/process_rolling_logger.py

```diff
@@ -0,0 +1,473 @@
+from whylogs.api.logger.experimental.logger.actor.process_actor import ProcessActor
+from whylogs.api.logger.experimental.logger.actor.process_rolling_logger_messages import (
+    DataDict,
+    FlushMessage,
+    LogEmbeddingRequestDict,
+    LogMessage,
+    LogRequestDict,
+    ProcessLoggerStatus,
+    ProcessLoggerStatusMessage,
+    RawLogEmbeddingsMessage,
+    RawLogMessage,
+    RawPubSubEmbeddingMessage,
+    RawPubSubMessage,
+    data_dict_from_pandas,
+    determine_dataset_timestamp,
+    get_columns,
+    log_dict_to_data_frame,
+    log_dict_to_embedding_matrix,
+    reduce_embeddings_request,
+    reduce_log_requests,
+)
+from whylogs.api.logger.experimental.logger.actor.string_util import encode_strings
+from whylogs.api.logger.experimental.logger.actor.thread_rolling_logger import (
+    LoggerStatus,
+    StatusMessage,
+    ThreadRollingLogger,
+)
+from whylogs.api.logger.experimental.logger.actor.time_util import (
+    Schedule,
+    TimeGranularity,
+    current_time_ms,
+)
+from whylogs.api.whylabs.session.session_manager import get_current_session
+from whylogs.api.writer import Writer, Writers
+from whylogs.core.schema import DatasetSchema
+from whylogs.core.stubs import pd
+
+MessageType = Union[
+    FlushMessage,
+    RawLogMessage,
+    RawLogEmbeddingsMessage,
+    RawPubSubMessage,
+    RawPubSubEmbeddingMessage,
+    LogMessage,
+    CloseMessage,
+    ProcessLoggerStatusMessage,
+]
+
+DataTypes = Union[str, int, float, bool, List[float], List[int], List[str]]
+
+DictType = TypeVar("DictType", bound="Union[LogRequestDict, LogEmbeddingRequestDict]")
+Loggable = Union[pd.DataFrame, Dict[str, Any]]
+
+
+class ProcessRollingLogger(ProcessActor[MessageType], DataLogger[Dict[str, ProcessLoggerStatus]]):
+    def __init__(
+        self,
+        aggregate_by: TimeGranularity = TimeGranularity.Hour,
+        write_schedule: Optional[Schedule] = Schedule(cadence=TimeGranularity.Minute, interval=10),
+        schema: Optional[DatasetSchema] = None,
+        sync_enabled: bool = False,
+        current_time_fn: Optional[Callable[[], int]] = None,
+        queue_config: QueueConfig = QueueConfig(),
+        thread_queue_config: QueueConfig = QueueConfig(),
+        debug_writers: List[Writer] = [],
+    ) -> None:
+        super().__init__(queue_config=queue_config)
+        self._sync_enabled = sync_enabled
+        self._thread_queue_config = thread_queue_config
+        self.current_time_ms = current_time_fn or current_time_ms
+        self.loggers: Dict[str, ThreadRollingLogger] = {}
+        self.debug_writers = debug_writers
+        self.write_schedule = write_schedule
+        self.schema = schema
+        self.aggregate_by = aggregate_by
+        self._pipe_signaler: Optional[PipeSignaler] = PipeSignaler() if sync_enabled else None
+        self._session = get_current_session()
+
+    def create_writers(self, dataset_id: str) -> List[Writer]:
```

Comment:
shouldn't it be possible to create_writers to other locations? and have WhyLabs as default